src/attr.rs

```diff
@@ -78,58 +80,49 @@ pub(crate) fn parse_our_attr(attr: &Attribute) -> Result<OurAttr, ()> {
     let name = match it.next() {
         Some(TokenTree::Ident(x)) => x,
         Some(other) => {
-            return Spanned::span(&other)
-                .err(format!("expected ident, found '{}'", other))
-                .emit_with_attr_note();
+            emit_error!(other.span(), "expected ident, found '{}'", other);
+            return Err(());
         }
         None => {
-            return attr.tokens.span()
-                .err("expected ident, found nothing")
-                .emit_with_attr_note();
+            emit_error!(attr.tokens.span(), "expected ident, found nothing");
+            return Err(());
         }
     };
 
-    // Extract the parameters (which again, have to be a group delimitted by
+    // Extract the parameters (which again, have to be a group delimited by
     // `()`)
     let params = match it.next() {
-        Some(TokenTree::Group(ref g)) if g.delimiter() == Delimiter::Parenthesis => {
-            g.stream()
-        }
+        Some(TokenTree::Group(ref g)) if g.delimiter() == Delimiter::Parenthesis => g.stream(),
         Some(other) => {
-            let msg = format!(
+            emit_error!(
+                other.span(),
                 "expected arguments for '{}' in parenthesis `()`, found `{}`",
                 name,
-                other,
+                other
             );
-            return Spanned::span(&other)
-                .err(msg)
-                .emit_with_attr_note();
+            return Err(());
         }
         None => {
-            let msg = format!(
+            emit_error!(
+                body.span(),
                 "expected arguments for '{}' in parenthesis `()`, found nothing",
-                name,
+                name
             );
-            return body.span()
-                .err(msg)
-                .emit_with_attr_note();
+            return Err(());
         }
     };
 
     // Finally match over the name of the attribute.
-    let out = match () {
-        () if name == "keep_default_for" => {
-            let proxy_types = parse_types(params.into())?;
-            OurAttr::KeepDefaultFor(proxy_types)
-        }
-        _ => {
-            return Spanned::span(&name)
-                .err(format!("invalid attribute '{}'", name))
-                .emit_with_attr_note();
-        }
-    };
```

Comment:
Out of curiosity, could you explain why the hell you did `match () { () if guard }` here?

Reply:
I expected that we would add a bunch more attributes. In that case a match would be better. But yeah, with a single arm (plus default) it's a bit strange. Also, `match name.to_string()` is probably better anyway, since the allocation probably really does not matter performance-wise. Anyway, gone now.

Reply:
Just a thing to consider: I don't think that 
```
match () {
    () if cond1 => {},
    () if cond2 => {},
    // ...
}
```

is any better than 
```rust
if cond1 {
   // ...
} else cond2 {
  // ...
} else {
  //  ...
}
```
or 
```rust
match &*ident.to_string() {
    "foo" => {},
    "bar" => {}
}
```

And don't be concerned by an extra allocation in the last snippet: *all* of those `to_string()` allocation take less than 1% in a typical proc-macro